convert_checkpoints.py

```diff
@@ -229,6 +235,9 @@ def _merge_weights(checkpoints, minimize_memory_footprint):
                 f'Tensors must be identical across shards for {key}'
             )
           state_dict_for_key[key] = tensors[0]
+
+        if enable_float32:
```

Comment:
In general, branch prediction should happen at the outer most loop. Putting `if` condition inside is very expensive. Thanks!

Reply:
No, current approach is already most efficient. There are another condition check inside the loop. If put if outside, the code readiness much worse than current one. On the performance side, both OPs are exactly same. For n condition check and 1 iteration of size k element, doesn't what way you write code, the time complexity is O(K*N).